src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -316,7 +316,8 @@ public void handleEvent(@NonNull DcEvent event) {
     int eventId = event.getId();
     if ((eventId == DcContext.DC_EVENT_WEBXDC_STATUS_UPDATE && event.getData1Int() == dcAppMsg.getId())) {
       Log.i(TAG, "handleEvent");
-      webView.loadUrl("javascript:document.getElementById('frame').contentWindow.__webxdcUpdate();");
+      final String window = internetAccess? "window" : "document.getElementById('frame').contentWindow";
+      webView.loadUrl("javascript:" + window + ".__webxdcUpdate();");
```

Comment:
even if slightly longer, i fine the following easier to read. 

but treat this just as a opinionated suggestion, it's up to you, i wold be fine with both, no blocker.

```suggestion
      if (internetAccess) {
        webView.loadUrl("javascript:window.__webxdcUpdate();");
      } else {
        webView.loadUrl("javascript:document.getElementById('frame').contentWindow.__webxdcUpdate();");
      }
```